dist/performance.js

```diff
@@ -227,60 +239,141 @@ function calcOcclusion(rect){
     return Math.max(0, calcArea(clipped_rect)/calcArea(rect))
 }
 
+function splitSrcSet(srcset) {
+    // "img.jpg 100w, img2.jpg 300w"
+    return srcset.split(',').map(srcDesc => {
+        // "img.jpg 100w", " img2.jpg 300w"
+        const src = srcDesc.trim().split(' ')[0];
+        return new URL(src, location.href).href;
+    });
+}
+
+function parseLinkHeader(link) {
+    if (!link) {
+        return {};
+    }
+
+    const srcPattern = /<([^>]+)>/;
+    const paramPattern = /([^=]+)=['"]?([^'"]+)['"]?/;
+    return Object.fromEntries(link.split(',').map(l => {
+        let [src, ...params] = l.split(';');
+
+        if (!srcPattern.test(src)) {
+            return [];
+        }
+        src = src.match(srcPattern)[1];
+        src = new URL(src, location.href).href;
+
+        params = params.map(p => {
+            p = p.trim().toLowerCase();
+            if (!paramPattern.test(p)) {
+                return;
+            }
+            const [_, key, value] = p.match(paramPattern);
+            return {key, value};
+        });
+        return [src, params];
+    }));
+}
+
+function getLcpPreloadInfo(rawDoc, lcpUrl) {
+    const lcpPreload = {};
+
+    const preloadTag = Array.from(rawDoc.querySelectorAll('link')).find(link => {
+        if (link.rel != 'preload') {
+            return false;
+        }
+
+        let src = link.href;
+        if (link.hasAttribute('imagesrcset')) {
+            src = splitSrcSet(link.imagesrcset).find(src => src == lcpUrl);
+        }
+
+        return src == lcpUrl;
+    });
+
+    if (preloadTag) {
+        lcpPreload.tag = summarizeElement(preloadTag);
+    }
+
+    const linkHeaderString = response_bodies[0].response_headers.link;
```

Comment:
I'm not sure what the format of the `response_headers` object is, but it's definitely possible for a page to contain more than 1 `Link` header, so it's probably worth double-checking that this data structure supports that case.

Reply:
This comment was resolved, but can you confirm that this is supported?

I'm thinking about a case where a page responds with multiple `Link: ...` headers:

```
Link: <script1.js>;rel="preload";as="script"
Link: <script2.js>;rel="preload";as="script"
```

Does the `response_headers` object merge those into a single value?

Reply:
Yes it does.

For example alistpart.com has the following:

<img width="673" alt="image" src="https://user-images.githubusercontent.com/10931297/176252562-8a891bed-e771-4e5e-8b0a-1386cf0d526e.png">

And using this, simplified, custom-metric:

```js
[link-headers]

function parseLinkHeader(link) {
    if (!link) {
        return {};
    }

    const srcPattern = /<([^>]+)>/;
    const paramPattern = /([^=]+)=['"]?([^'"]+)['"]?/;
    return Object.fromEntries(link.split(/[,\n]/).map(l => {
        let [src, ...params] = l.split(';');

        if (!srcPattern.test(src)) {
            return [];
        }
        src = src.match(srcPattern)[1];
        src = new URL(src, location.href).href;

        params = params.map(p => {
            p = p.trim().toLowerCase();
            if (!paramPattern.test(p)) {
                return;
            }
            const [_, key, value] = p.match(paramPattern);
            return {key, value};
        });
        return [src, params];
    }));
}

const response_bodies = $WPT_BODIES;
return parseLinkHeader(response_bodies[0].response_headers.link);
```

We see the following [in this example test](https://www.webpagetest.org/result/220628_AiDcRT_ERX/2/details/#waterfall_view_step1):

```json
{
    "https:\/\/alistapart.com\/wp-json\/": [
        {
            "key": "rel",
            "value": "https:\/\/api.w.org\/"
        }
    ],
    "https:\/\/alistapart.com\/wp-json\/wp\/v2\/pages\/6000475": [
        {
            "key": "rel",
            "value": "alternate"
        },
        {
            "key": "type",
            "value": "application\/json"
        }
    ],
    "https:\/\/wp.me\/PaHlEW-paZR": [
        {
            "key": "rel",
            "value": "shortlink"
        }
    ]
}
```

So you can see it picked up all three link headers and `parseLinkHeader` correctly separated them out.

Reply:
Great, thanks for confirming!